ledger/participant-state/kvutils/src/main/protobuf/daml_kvutils.proto

```diff
@@ -408,10 +336,30 @@ message DamlCommandDedupKey {
   string application_id = 2;
   string command_id = 3;
 }
+
 message DamlCommandDedupValue {
-  // NOTE(JM): Currently no content. Could store pointer to log entry.
+  // Currently no content. Could store pointer to log entry.
 }
 
+message DamlSubmissionDedupKey {
+  enum SubmissionKind {
+    PARTY_ALLOCATION = 0;
+    PACKAGE_UPLOAD = 1;
+    CONFIGURATION = 2;
+  }
+
+  SubmissionKind submission_kind = 1;
+
+  // A unique string scoped to a particular participant.
+  string submission_id = 2;
+
+  // Uploading participant's id.
+  string participant_id = 3;
+}
+
+message DamlSubmissionDedupValue {
+  // Currently no content. Could store pointer to log entry.
```

Comment:
We should add some content here as e.g. Sawtooth doesn't like empty values.

Reply:
Done, added recordTime as value throughout